src/app/shared/http.service.ts

```diff
@@ -46,12 +46,21 @@ export class HttpService extends Http {
 
   private catchRequestError () {
     return (res: Response) => {
-      if (res.status === 403 || ( res.status === 401 && !res.headers.has("www-authenticate"))) {
+      if (res.status === 403 || this.isAuthenticationError(res)) {
```

Comment:
In case of missed or invalid GitHub or OSO tokens we return "unauthorized_error" error code (via JSON error) + plus "WWW-Authenticate: LINK" header. So, checking just 'jwt_security_error' code without checking "WWW-Authenticate" header should be fine for now.

Reply:
@alexeykazakov making a note on a PR will get lost. Do you want this done now or do you want to make an issue to track it?

Reply:
@joshuawilson , since we are already checking if the 401 is associated with a jwt error, Alexey's comments are already addressed in the PR.